test/create-request-client.js

```diff
@@ -534,4 +534,117 @@ describe('request client', () => {
       });
     });
   });
+
+  describe('resolves body and header curlies', () => {
+    it('should keep valid data types', () => {
+      const event = {
+        bundle: {
+          inputData: {
+            number: 123,
+            bool: true,
+            float: 123.456,
+            arr: [1, 2, 3]
+          }
+        }
+      };
+      const bodyInput = createInput({}, event, testLogger);
+      const request = createAppRequestClient(bodyInput);
+      return request({
+        url: 'http://zapier-httpbin.herokuapp.com/post',
+        method: 'POST',
+        body: {
+          number: '{{bundle.inputData.number}}',
+          bool: '{{bundle.inputData.bool}}',
+          float: '{{bundle.inputData.float}}',
+          arr: '{{bundle.inputData.arr}}'
+        }
+      }).then(response => {
+        const { json } = response.json;
+
+        json.number.should.eql(123);
+        json.bool.should.eql(true);
+        json.float.should.eql(123.456);
+        json.arr.should.eql([1, 2, 3]);
+      });
+    });
+
+    it('should interpolate strings', () => {
+      const event = {
+        bundle: {
+          inputData: {
+            resourceId: 123
+          },
+          authData: {
+            access_token: 'Let me in'
+          }
+        }
+      };
+      const bodyInput = createInput({}, event, testLogger);
+      const request = createAppRequestClient(bodyInput);
+      return request({
+        url: 'http://zapier-httpbin.herokuapp.com/post',
+        method: 'POST',
+        body: {
+          message: 'We just got #{{bundle.inputData.resourceId}}'
+        },
+        headers: {
+          Authorization: 'Bearer {{bundle.authData.access_token}}'
+        }
+      }).then(response => {
+        const { json, headers } = response.json;
+
+        json.message.should.eql('We just got #123');
+        headers.Authorization.should.eql('Bearer Let me in');
+      });
+    });
+
+    it('should throw when interpolating a string with an array', () => {
+      const event = {
+        bundle: {
+          inputData: {
+            badData: [1, 2, 3]
+          }
+        }
+      };
+      const bodyInput = createInput({}, event, testLogger);
+      const request = createAppRequestClient(bodyInput);
+      return request({
+        url: 'http://zapier-httpbin.herokuapp.com/post',
+        method: 'POST',
+        body: {
+          message: 'No arrays, thank you: {{bundle.inputData.badData}}'
+        }
+      }).should.be.rejectedWith(
+        "Cannot interpolate objects or arrays into a string. You've sent an Array, which will get coerced to 1,2,3"
```

Comment:
> coerced to 1,2,3

I find this wording a little confusing, since it's not clear that `1,2,3` is a string. Maybe change the error to be 

`which will get coerced into the string "1,2,3"`

Reply:
also maybe change it to `Cannot reliably interpolate...` to make it clear that we _can_, but it might have weird behavior.

Reply:
Good points

Reply:
Is `which will get coerced into the string` actually true? We are always going to throw this error, yea? Maybe just a stringified dump of the first 10-20 characters to help them know _which_ array we are getting?

Reply:
Yeah it will always throw. The intent was to say, "Hey there, here's what would have gone through". But maybe that's unnecessary. We could just say, "Can't do it. Here's what you sent".

We are still sending them the data above. But I can see the value in keeping the output simpler.